concept/type_/type_cache.rs

```diff
@@ -608,6 +608,55 @@ impl TypeCache {
         let as_u16 = type_vertex.type_id_().as_u16();
         attribute_type_caches[as_u16 as usize].as_ref()
     }
+
+
+    pub(crate) fn __generalised__get_supertype<'a, T>(&self, type_: T) -> Option<T::CacheParameter>
+    where T: TypeAPI<'a> + CommonTypeCacheSelector
+    {
+        // TODO: Why does this not return &Option<EntityType<'static>> ?
+        // TODO: All the unwrap
+        Some(T::select(self, type_).supertype.as_ref()?.clone())
+    }
+
+    pub(crate) fn __generalised__get_label<'a, T>(&self, type_: T) -> &Label<'static>
+    where T: TypeAPI<'a> + CommonTypeCacheSelector,
+          <T as CommonTypeCacheSelector>::CacheParameter: 'static
+    {
+        &T::select(self, type_).label
+    }
```

Comment:
These can replace the code above that's repeated for each of EntityType, RelationType, AttributeType, RoleType